#====================
#====================

# shell script for obtaining and aligning ChIP-seq data from ENCODE, GSE31477. Raw data generated by: Consortium EP, (2011). ENCODE Project Consortium. A userâ€™s guide to the encyclopedia of DNA elements (ENCODE). PLoS Biol. 9, e1001046).

# Required:
#1. appropriate reference genome, generated with: bowtie2-build 
#2. fastx_tools     - can be downloaded from http://hannonlab.cshl.edu/fastx_toolkit/download.html
#3. samtools - can be downloaded from http://www.htslib.org/download/
#4. bowtie2 - can be downlaoded from http://bowtie-bio.sourceforge.net/bowtie2/index.shtml
#5. MACS3 - can be obtained fom: https://github.com/macs3-project/MACS
#6. fastq_dump of sra-tools - https://ncbi.github.io/sra-tools/fastq-dump.html
                                                        
#=============================================================================================

##Paths:
### update the path to the folder containing the bt2 genome built.

genomehg38="/PATH/hg38"

#=============================================================================================

### Obtaining the datasets:

#############################################################
#https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE31477
#############################################################




fastq-dump SRR502172
fastq-dump SRR502173
cat SRR502172.fastq SRR502173.fastq > K562_NELFe.fastq

fastq-dump SRR502177
fastq-dump SRR502178
cat SRR502177.fastq SRR502178.fastq > K562_GTF2B.fastq

fastq-dump SRR502192
fastq-dump SRR502193
cat SRR502192.fastq SRR502193.fastq > K562_p300.fastq

fastq-dump SRR502023
fastq-dump SRR502024
cat SRR502023.fastq SRR502024.fastq > K562_RAD21.fastq

fastq-dump SRR502109
fastq-dump SRR502110
cat SRR502109.fastq SRR502110.fastq > K562_Input.fastq

fastq-dump SRR502378
fastq-dump SRR502379
cat SRR502379.fastq SRR502378.fastq > K562_TBP.fastq

fastq-dump SRR568133
fastq-dump SRR568134
cat SRR568133.fastq SRR568134.fastq > K562_GATA1.fastq

fastq-dump SRR502204
fastq-dump SRR502205
cat SRR502205.fastq SRR502204.fastq > K562_CTCF.fastq

fastq-dump SRR227511
fastq-dump SRR227512
cat SRR227511.fastq SRR227512.fastq > K562_H3K36me3.fastq


rm SRR*.fastq



List="K562_NELFe K562_GTF2B K562_p300 K562_RAD21 K562_Input K562_TBP K562_GATA1 K562_CTCF K562_H3K36me3"

for x in ${List}

do

echo ${x}

    fastq_quality_filter -q 20 -p 80 -v -i ${x}.fastq -o ${x}_qfilt.fastq
    
    bowtie2 -p 7 -x ${genomehg38} -U ${x}_qfilt.fastq | awk '$2 != 4 {print}' | sed '/XS:/d' | samTools view -S -b '-' > ${x}.bam

    ### please note that only reads that uniquely map to hg38 are retained.
    
done


factList="K562_NELFe K562_GTF2B K562_p300 K562_RAD21 K562_TBP K562_GATA1 K562_CTCF"

ctrl=K562_Input.bam

for x in ${factList}
do
    macs3 callpeak -t ${x}.bam -c $ctrl -f BAM -g hs -B --call-summits -n ${x}
done


histList="K562_H3K36me3"

for x in ${histList}
do
    macs3 callpeak -t ${x}.bam -c $ctrl -f BAM -g hs -B -n ${x}
done


